File: VB/MyWizardCustomizationService.vb
Imports DevExpress.Xpf.Reports.UserDesigner.ReportWizard
Imports DevExpress.DataAccess.Wizard.Model
Imports DevExpress.Xpf.DataAccess.DataSourceWizard
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Wizards
Imports DevExpress.DataAccess.Wizard
Imports System.Drawing
Imports System.Runtime.InteropServices

Namespace SingleQueryWizard

    Public Class MyWizardCustomizationService
        Implements IWizardCustomizationService

        Private Sub CustomizeDataSourceWizard(ByVal customization As DataSourceWizardCustomizationModel, ByVal container As ViewModelSourceIntegrityContainer) Implements IDataSourceWizardCustomizationService.CustomizeDataSourceWizard
            Dim wizardOptions = container.Resolve(Of SqlWizardOptions)()
            wizardOptions = wizardOptions And Not SqlWizardOptions.MultiQueryWizard 'or you can use 'var wizardOptions = SqlWizardOptions.None'
            container.RegisterInstance(wizardOptions)
        End Sub

        Private Sub CustomizeReportWizard(ByVal customization As ReportWizardCustomizationModel, ByVal container As ViewModelSourceIntegrityContainer) Implements IWizardCustomizationService.CustomizeReportWizard
            Dim wizardOptions = container.Resolve(Of SqlWizardOptions)()
            wizardOptions = wizardOptions And Not SqlWizardOptions.MultiQueryWizard 'or you can use 'var wizardOptions = SqlWizardOptions.None'
            container.RegisterInstance(wizardOptions)
        End Sub

        Private Function TryCreateDataSource(ByVal model As IDataSourceModel, <Out> ByRef dataSource As Object, <Out> ByRef dataMember As String) As Boolean Implements IDataSourceWizardCustomizationService.TryCreateDataSource
            dataSource = Nothing
            dataMember = Nothing
            Return False
        End Function

        Private Function TryCreateReport(ByVal model As XtraReportModel, <Out> ByRef report As XtraReport) As Boolean Implements IWizardCustomizationService.TryCreateReport
            report = Nothing
            Return False
        End Function
    End Class
End Namespace

File: CS/MyWizardCustomizationService.cs
using DevExpress.Xpf.Reports.UserDesigner.ReportWizard;
using DevExpress.DataAccess.Wizard.Model;
using DevExpress.Xpf.DataAccess.DataSourceWizard;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Wizards;
using DevExpress.DataAccess.Wizard;
using System.Drawing;

namespace SingleQueryWizard {
    public class MyWizardCustomizationService : IWizardCustomizationService {
        void IDataSourceWizardCustomizationService.CustomizeDataSourceWizard(DataSourceWizardCustomizationModel customization, ViewModelSourceIntegrityContainer container) {
            var wizardOptions = container.Resolve<SqlWizardOptions>();
            wizardOptions &= ~SqlWizardOptions.MultiQueryWizard; //or you can use 'var wizardOptions = SqlWizardOptions.None'
            container.RegisterInstance(wizardOptions);
        }

        void IWizardCustomizationService.CustomizeReportWizard(ReportWizardCustomizationModel customization, ViewModelSourceIntegrityContainer container) {
            var wizardOptions = container.Resolve<SqlWizardOptions>();
            wizardOptions &= ~SqlWizardOptions.MultiQueryWizard; //or you can use 'var wizardOptions = SqlWizardOptions.None'
            container.RegisterInstance(wizardOptions);
        }

        bool IDataSourceWizardCustomizationService.TryCreateDataSource(IDataSourceModel model, out object dataSource, out string dataMember) {
            dataSource = null;
            dataMember = null;
            return false;
        }

        bool IWizardCustomizationService.TryCreateReport(XtraReportModel model, out XtraReport report) {
            report = null;
            return false;
        }
    }
}
